packages/core/src/index.ts

```diff
@@ -135,14 +145,58 @@ export function buildSendMail<T>(options: BuildSendMailOptions<T>) {
         error(`Caught error ${e}`);
         error("Is the mailing preview server running?");
       }
-
       return;
     }
 
-    const response = await options.transport.sendMail(htmlMail);
+    if (analyticsEnabled) {
+      const hookResponse = await fetch(MAILING_API_URL + "/api/messages", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          "x-api-key": MAILING_API_KEY,
+        },
+        body: JSON.stringify({
+          anonymousId,
+          templateName: templateName || derivedTemplateName,
+          previewName,
+          mailOptions,
+        }),
+      });
+
+      if (hookResponse.status === 200) {
+        const {
+          message: { id: messageId },
+        } = await hookResponse.json();
+        const stringHtml = mailOptions.html?.toString();
+        if (stringHtml) {
+          mailOptions.html = instrumentHtml({
+            html: stringHtml,
+            messageId: messageId,
+            apiUrl: MAILING_API_KEY,
+          });
+        }
+      } else {
+        const json = await hookResponse.json();
```

Comment:
should decode json in a try/catch in case it's a 500 and it throws a parsing error